### PR TITLE
Added severity levels and profiles of yet uncategorized checkers

### PR DESCRIPTION
**Pull request by dkrupp**
New profiles and number of findings in open source projects in ()
default:
alpha.cplusplus.UninitializedObject (0)
cplusplus.InnerPointer(0) 
bugprone-copy-constructor-init (0)
bugprone-narrowing-conversions (75)
bugprone-unused-return-value (7)
bugprone-terminating-continue (1)
bugprone-throw-keyword-missing (0)
bugprone-exception-escape (5)
bugprone-virtual-near-miss (1)
optin.cplusplus.VirtualCall(98)
cert-fio38-c (0) // do not copy a FILE object.

sensitive:
bugprone-macro-parentheses (621)
alpha.security.MmapWriteExec (0)
cert-msc50-cpp (0)
cert-msc51-cpp(0)
cert-dcl54-cpp(3) //Allocation and deallocation functions can be overloaded at both global and class scopes.
cert-err09-cpp (13) //Finds violations of the rule “Throw by value, catch by reference” presented for example in “C++ Coding Standards” 
by H. Sutter and A. Alexandrescu.

extreme:
misc-unused-parameters (2259)
security.insecureAPI.bcmp //not security but obsolete API
security.insecureAPI.bcopy //not security but obsolete API
security.insecureAPI.bzero //not security but obsolete API
security.insecureAPI.strcpy //not security but obsolete API
optin.performance.Padding (35) 

security:
cert-*
alpha.security.*
security.*

Resolves: #1054

### CODE REVIEW

File: config/checker_severity_map.json [hunk 1]
```diff
@@ -1,4 +1,5 @@
 {
+  "abseil-string-find-startswith":                               "STYLE",
```
Comment (csordasmarton) [inline config/checker_severity_map.json]: Indentation:
```suggestion
  "abseil-string-find-startswith":                              "STYLE",
```

File: config/checker_severity_map.json [hunk 2]
```diff
@@ -14,6 +15,7 @@
   "alpha.cplusplus.DeleteWithNonVirtualDtor":                   "HIGH",
   "alpha.cplusplus.IteratorRange":                              "MEDIUM",
   "alpha.cplusplus.MisusedMovedObject":                         "MEDIUM",
+  "alpha.cplusplus.UninitializedObject":                         "MEDIUM",
```
Comment (csordasmarton) [inline config/checker_severity_map.json]: Indentation:
```suggestion
  "alpha.cplusplus.UninitializedObject":                        "MEDIUM",
```

File: config/checker_severity_map.json [hunk 3]
```diff
@@ -288,11 +307,15 @@
...
   "readability-simplify-boolean-expr":                          "MEDIUM",
+  "readability-simplify-subscript-expr":                        "STYLE",
   "readability-static-accessed-through-instance":               "STYLE",
   "readability-static-definition-in-anonymous-namespace":       "STYLE",
   "readability-string-compare":                                 "LOW",
   "readability-uniqueptr-delete-release":                       "STYLE",
   "security.FloatLoopCounter":                                  "MEDIUM",
+  "security.insecureAPI.bcmp":                                  "LOW",
+  "security.insecureAPI.bcopy":                                 "LOW",
+  "security.insecureAPI.bzero":                                  "LOW",
```
Comment (csordasmarton) [inline config/checker_severity_map.json]: Indentation:
```suggestion
  "security.insecureAPI.bzero":                                 "LOW",
```

File: config/config.json [hunk 4]
```diff
@@ -12,7 +12,8 @@
       "default" : "High-quality standard checks with a low false positive rate.",
       "sensitive" : "Default checks + more comprehensive checks with a low false positive rate.",
       "extreme" : "Sensitive checks + more comprehensive checks with a manageable false positive rate.",
-      "portability" : "Checks that aim to detect code issues emerging from platform differences (eg. between 32-bit and 64-bit architectures)."
+      "portability" : "Checks that aim to detect code issues emerging from platform differences (eg. between 32-bit and 64-bit architectures).",
+      "security" : "Checkers that warn for potentially vulnerable code"
```
Comment (csordasmarton) [inline config/config.json]:
```suggestion
      "security" : "Checkers that warn for potentially vulnerable code."
```

File: config/config.json [hunk 5]
```diff
@@ -27,13 +28,15 @@
         "alpha.cplusplus.DeleteWithNonVirtualDtor" : ["default", "sensitive", "extreme"],
         "alpha.cplusplus.IteratorRange" : [],
         "alpha.cplusplus.MisusedMovedObject" : ["sensitive", "extreme"],
-        "alpha.security.ArrayBound" : ["extreme"],
-        "alpha.security.ArrayBoundV2" : ["extreme"],
-        "alpha.security.MallocOverflow" : ["sensitive", "extreme"],
-        "alpha.security.ReturnPtrRange" : ["sensitive", "extreme"],
-        "alpha.security.taint.TaintPropagation" : ["sensitive", "extreme"],
+        "alpha.cplusplus.UninitializedObject" : ["default", "sensitive", "extreme"],
+        "alpha.security.ArrayBound" : ["extreme","security"],
```
Comment (csordasmarton) [inline config/config.json]: Please put a space between a comma and a quotation mark in every place similar to this:
```suggestion
        "alpha.security.ArrayBound" : ["extreme", "security"],
```

File: config/checker_severity_map.json [hunk 6]
```diff
@@ -55,6 +59,7 @@
   "bugprone-misplaced-widening-cast":                           "HIGH",
   "bugprone-move-forwarding-reference":                         "MEDIUM",
   "bugprone-multiple-statement-macro":                          "MEDIUM",
+  "bugprone-narrowing-conversions":                             "MEDIUM",
```
Comment (gyorb) [inline config/checker_severity_map.json]: will be renamed in clang9, I did not found `bugprone-narrowing-conversions` only `cppcoreguidelines-narrowing-conversions`

File: config/config.json [hunk 7]
```diff
@@ -93,7 +103,9 @@
         "bugprone-argument-comment" : ["sensitive", "extreme"],
         "bugprone-assert-side-effect" : ["default", "sensitive", "extreme"],
         "bugprone-bool-pointer-implicit-conversion" : ["default", "sensitive", "extreme"],
+        "bugprone-copy-constructor-init" : ["default", "sensitive", "extreme"],
         "bugprone-dangling-handle" : ["default", "sensitive", "extreme"],
+        "bugprone-exception-escape" : ["default", "sensitive", "extreme"],
```
Comment (gyorb) [inline config/config.json]: Are we sure we want to add this to the default set?
There is a warning for this checker `WARNING! This check may be expensive on large source files.`
https://clang.llvm.org/extra/clang-tidy/checks/bugprone-exception-escape.html

Comment (dkrupp) [reply config/config.json]: put to sensitive, extreme only. We will need to measure this to put into the default profile.

File: config/checker_severity_map.json (hunk 6 above)
Comment (dkrupp) [reply config/checker_severity_map.json]: put to cppcoreguidelines only, and only to extreme level.

**Pull request merged**